lib/services/build.ts

```diff
@@ -444,14 +453,19 @@ export class BuildService implements Project.IBuildService {
 				device: device
 			}).wait();
 
-			let result = _.filter(buildResult, (def: Server.IPackageDef) => !def.disposition || def.disposition === "BuildResult")[0].localFile;
-			return result;
-		}).future<string>()();
+			let packageName = _.filter(buildResult, (def: Server.IPackageDef) => !def.disposition || def.disposition === "BuildResult")[0].localFile;
+			let metadata = _.filter(buildResult, (def: Server.IPackageDef) => !def.disposition || (def.disposition === "BuildResultMetadata" && def.key === "AppIdentifier"))[0];
```

Comment:
I do believe that this would be better:
```TypeScript
let metadata = _.filter(buildResult, (def: Server.IPackageDef) => def.disposition === "BuildResultMetadata" && def.key === "AppIdentifier")[0];
```

Since you're only actually interested in the `AppIdentifier` itself.